index.js

```diff
@@ -267,12 +267,12 @@ function buildRouteCompress (fastify, params, routeOptions, decorateOnly) {
         encoding === undefined
           ? reply.removeHeader('Content-Encoding')
           : reply.header('Content-Encoding', 'identity')
-        pump(stream, payload = unzipStream(params.uncompressStream), onEnd.bind(reply))
+        pipeline(stream, payload = unzipStream(params.uncompressStream), onEnd.bind(reply))
       }
       return next(null, payload)
     }
 
-    if (typeof payload.pipe !== 'function') {
+    if (typeof payload.pipe !== 'function' && !(payload instanceof ReadableStream)) {
```

Comment:
Unfortunately using `instanceof` here would break compatibility everywhere.

Reply:
yeah I suspected as such. 
I think solving the linked issue (#309) probably requires more significant re-architecture of this plugin to more modern standards (as you discussed also here https://github.com/fastify/fastify-compress/issues/297#issuecomment-2119902305) which I don't have the time/ability to work on right now, so I will close this PR, though i'll leave my branch here as the test case may be useful to anyone else who might want to pick this up